lib/IDE/CodeCompletionResultBuilder.h

```diff
@@ -64,8 +64,10 @@ class CodeCompletionResultBuilder {
 public:
   CodeCompletionResultBuilder(CodeCompletionResultSink &Sink,
                               CodeCompletionResultKind Kind,
-                              SemanticContextKind SemanticContext)
-      : Sink(Sink), StringBuilder(*Sink.Allocator, Sink.annotateResult),
+                              SemanticContextKind SemanticContext,
+                              const DeclContext *CurrDeclContext = nullptr)
```

Comment:
Hmm, I don't love this being an optional parameter since methods like `addTypeAnnotation` will crash without it, and that relationship isn't obvious. I think either this should be a mandatory parameter for `CodeCompletionResultBuilder`, or we should just make it a parameter for the 3 methods that need it. I think I'm leaning towards the latter.

Reply:
I think the latter makes more sense too. I've updated all methods to take a `DeclContext&` instead to make sure it cannot be null since as you mentioned it'd crash without it anyway.